### PR TITLE
Update Azure.Batch to MIT license.

### PR DESCRIPTION
**Pull request by matthchr**
- Also updated readme.md and changelog.md in preparation for this to be released.
  - Includes update to the nuget package license as well.

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [ ] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by matthchr** [comment]
@hovsepm - Here you go

@darylmsft @itowlson @xingwu1 - please review

### Before merging
Please wait until some folks from my team review this and sign off

File: src/Batch/Client/README.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# License notes
+The Azure Batch C# client is now under the MIT license. Prior to 3/10/2017 it was under the Apache 2.0 license.
```
Comment (itowlson) [inline src/Batch/Client/README.md]: Prefer to spell out dates - many readers will be in dd/mm/yyyy locales.

File: src/Batch/Client/changelog.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 ## Azure.Batch release notes
 
+### Upcoming changes
+These changes are planned but haven't been published yet.
+
+#### License
+Moved source code and NuGet package from Apache 2.0 license to MIT license. This is more consistent with the other Azure SDKs as well as other open source projects from Microsoft such as DotNet.
```
Comment (itowlson) [inline src/Batch/Client/changelog.md]: "DotNet" is not normally spelled out that way.  The product is ".NET".  They use "dotnet" as the user name under which they keep their GitHub repos, and in other places where the leading "." isn't allowed... but this seems to refer to the product...?

File: src/Batch/Client/README.md (hunk 1 above)
Comment (matthchr) [inline src/Batch/Client/README.md]: Done

File: src/Batch/Client/changelog.md (hunk 2 above)
Comment (matthchr) [inline src/Batch/Client/changelog.md]: Done

**Pull request merged**